src/main/java/org/embulk/output/sftp/SftpUtils.java

```diff
@@ -0,0 +1,279 @@
+        }
+    }
+
+    public Void renameFile(final String before, final String after)
+    {
+        try {
+            return retryExecutor()
+                    .withRetryLimit(maxConnectionRetry)
+                    .withInitialRetryWait(500)
+                    .withMaxRetryWait(30 * 1000)
+                    .runInterruptible(new Retryable<Void>() {
+                        @Override
+                        public Void call() throws IOException
+                        {
+                            FileObject previousFile = manager.resolveFile(getSftpFileUri(before).toString(), fsOptions);
+                            FileObject afterFile = manager.resolveFile(getSftpFileUri(after).toString(), fsOptions);
+                            previousFile.moveTo(afterFile);
+                            logger.info("renamed remoted file: {} to {}", previousFile.getPublicURIString(), afterFile.getPublicURIString());
+
+                            return null;
+                        }
+
+                        @Override
+                        public boolean isRetryableException(Exception exception)
+                        {
+                            return true;
+                        }
+
+                        @Override
+                        public void onRetry(Exception exception, int retryCount, int retryLimit, int retryWait) throws RetryGiveupException
+                        {
+                            String message = String.format("SFTP rename remote file failed. Retrying %d/%d after %d seconds. Message: %s",
+                                    retryCount, retryLimit, retryWait / 1000, exception.getMessage());
+                            if (retryCount % 3 == 0) {
+                                logger.warn(message, exception);
+                            }
+                            else {
+                                logger.warn(message);
+                            }
+                        }
+
+                        @Override
+                        public void onGiveup(Exception firstException, Exception lastException) throws RetryGiveupException
+                        {
+                        }
+                    });
+        }
+        catch (RetryGiveupException ex) {
+            throw Throwables.propagate(ex.getCause());
+        }
+        catch (InterruptedException ex) {
+            throw Throwables.propagate(ex);
+        }
+    }
+
+    private URI getSftpFileUri(String remoteFilePath)
+    {
+        try {
+            return new URI("sftp", userInfo, host, port, remoteFilePath, null, null);
+        }
+        catch (URISyntaxException e) {
+            logger.error(e.getMessage());
+            throw new ConfigException(e);
+        }
+    }
+
+    private FileObject newSftpFile(final URI sftpUri) throws FileSystemException
+    {
+        FileObject file = manager.resolveFile(sftpUri.toString(), fsOptions);
+        if (file.exists()) {
```

Comment:
Minor: maybe it's nice to have a warn log, that file is going to be overwritten.

Reply:
Thanks. Fixed at 5e6e74f